src/Workspaces/CoreTestUtilities/MEF/FeaturesTestCompositions.cs

```diff
@@ -11,7 +11,11 @@ namespace Microsoft.CodeAnalysis.Test.Utilities
 {
     public static class FeaturesTestCompositions
     {
-        public static readonly TestComposition Features = TestComposition.Empty.AddAssemblies(MefHostServices.DefaultAssemblies);
-        public static readonly TestComposition RemoteHostFeatures = TestComposition.Empty.AddAssemblies(RoslynServices.RemoteHostAssemblies);
+        public static readonly TestComposition Features = TestComposition.Empty
+            .AddAssemblies(MefHostServices.DefaultAssemblies)
+            .AddParts(typeof(MockWorkspaceEventListenerProvider)); // by default, avoid running Solution Crawler and other services that start in workspace event listeners
```

Comment:
❔ Why do we need to avoid solution crawler in this case?

Reply:
Two reasons 
1) I don't think it's useful to run it for every single test that creates a workspace. 
2) Disposal is not working well in tests. 

We might want to fix [2]. Re [1] I think we should disable it by default and let the host enable it explicitly. It's odd that we would start running the crawler if someone hosts a workspace in their app.

```
System.AggregateException : One or more errors occurred.\r\n---- System.AggregateException : One or more errors occurred.\r\n-------- System.Threading.Tasks.TaskCanceledException : A task was canceled.\r\n---- System.TimeoutException : Failed to clean up listeners in a timely manner.\r\n  WorkItem WorkCoordinator.cs 473\r\n  WorkItem WorkCoordinator.LowPriorityProcessor.cs 98\r\n-------- System.OperationCanceledException : The operation was canceled.


Stack trace

----- Inner Stack Trace #1 (System.AggregateException) -----
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait(TimeSpan timeout)
   at Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService.WorkCoordinator.Shutdown(Boolean blockingShutdown) in /_/src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs:line 137
   at Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService.Unregister(Workspace workspace, Boolean blockingShutdown) in /_/src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerRegistrationService.cs:line 105
   at Microsoft.CodeAnalysis.SolutionCrawler.HostSolutionCrawlerWorkspaceEventListener.StopListening(Workspace workspace) in /_/src/Features/Core/Portable/SolutionCrawler/HostSolutionCrawlerWorkspaceEventListener.cs:line 34
   at Microsoft.CodeAnalysis.Host.DefaultWorkspaceEventListenerServiceFactory.Service.Stop() in /_/src/Workspaces/Core/Portable/Workspace/Host/EventListener/IWorkspaceEventListenerProvider.cs:line 90
   at Microsoft.CodeAnalysis.Workspace.Dispose(Boolean finalize) in /_/src/Workspaces/Core/Portable/Workspace/Workspace.cs:line 373
   at Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces.TestWorkspace.Dispose(Boolean finalize) in /_/src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace.cs:line 171
   at Microsoft.CodeAnalysis.Workspace.Dispose() in /_/src/Workspaces/Core/Portable/Workspace/Workspace.cs:line 359
   at Microsoft.CodeAnalysis.Editor.UnitTests.Formatting.InferredIndentationTests.<SingleLineWithFourSpaces>d__2.MoveNext()
```

Reply:
> I don't think it's useful to run it for every single test that creates a workspace (and it didn't before because it was only running for workspaces of kind "Host").

It's helpful for identifying problems related to it, and better represents the way the product is used in production. Ideally we could remove the override altogether.

> Disposal is not working well in tests.

I'll take a look at fixing this.

Reply:
> It's helpful for identifying problems related to it, and better represents the way the product is used in production.

I agree we should match the production. That's why I suggested the production code should not run solution crawler by default (in the Features layer). Instead, the host layer (i.e. VS, ServiceHub, or tests specifically targeting diagnostics) should do so.

I don't want all tests identifying problems with all components. That would not be unit testing anymore.